packages/core/src/pages/[...slug].tsx

```diff
@@ -91,12 +93,29 @@ export const getStaticProps: GetStaticProps<
   ]
 
   if (await landingPagePromise) {
+    // Checking if the fetch function corresponding to the slug exists
+    const fetchFunction = dynamicContent[slug]
+    let serverData = null
+
+    if (!fetchFunction) {
+      console.warn(`Warning: Fetch function not found for slug: ${slug}`)
+    } else {
+      // Calling the fetch function corresponding to the slug
+      const { data, errors = [] } = await fetchFunction()
+      serverData = data
+
+      if (errors.length > 0) {
+        console.error(...errors)
+      }
+    }
+
```

Comment:
suggestion: what do you think about creating a function with this snippet since it's so similar to the snippet on `/pages/index.tsx`?

Reply:
nice! done [here](https://github.com/vtex/faststore/pull/2242/commits/3729f2a4345ea526577ff2240a7f91c2d7e67da7).